test/unittests/libdevcore/core.cpp

```diff
@@ -54,4 +54,21 @@ BOOST_AUTO_TEST_CASE(isHex)
 	BOOST_CHECK(dev::isHex("003"));
 }
 
+BOOST_AUTO_TEST_CASE(integerToHex)
+{
+	using dev::toHex;
+	using dev::toCompactHex;
+	using dev::u256;
+	const auto n = u256{std::numeric_limits<int32_t>::max()};
+
+	// FIXME: Make it work for types other than u256.
+	BOOST_CHECK_EQUAL(toHex(u256{0x0}), "0");
```

Comment:
Honestly, no idea. I just added unit tests showing current behavior to make sure there is no functional changes in case these unused functions are removed.

Reply:
But I never was for printing integers as `0x00`. I believe "compact" vs "not-compact" affects only byte arrays.